storage/leveldb/leveldbSerial.go

```diff
@@ -319,6 +301,20 @@ func (s *SerialDB) DestroyClosed() error {
 	return err
 }
 
+// prepareClosingUnprotected will prepare the internal component for closing the database
+// must be called under mutex protection
+func (s *SerialDB) prepareClosingUnprotected() error {
```

Comment:
can be done in another PR: extract the common code in the level.go file also

Reply:
added a todo